README.md

```diff
@@ -1 +1,129 @@
+
+This implementation builds on [Lit-LLaMA](https://github.com/lightning-AI/lit-llama) and [nanoGPT](https://github.com/karpathy/nanoGPT), and it's powered by [Lightning Fabric](https://lightning.ai/docs/fabric/stable/) ⚡.
+
+Weights are available under the Apache 2.0 license and can be downloaded [following these instructions](howto/download_weights.md).
+
+## Design principles
+
+This repository follows the **Lit design principle** of **openness through clarity**.
+
+**Lit-StableLM** is:
+
+- **Simple:** Single-file implementation without boilerplate.
+- **Correct:** Numerically equivalent to the original model.
+- **Optimized:** Runs on consumer hardware or at scale.
+- **Open-source:** No strings attached.
+
+Avoiding code duplication is **not** a goal. **Readability** and **hackability** are.
+
+## Get involved!
+[Join our Discord](https://discord.gg/VptPCZkGNa) to build high-performance, truly open-source models for the common benefit of the community.
+
+&nbsp;
+
+## Setup
+
+Clone the repo
+
+```bash
+git clone https://github.com/Lightning-AI/lit-stablelm
+cd lit-llama
+```
+
+install dependencies
+
+```bash
+pip install -r requirements.txt
+```
+
+You are all set! 🎉
+
+&nbsp;
+
+## Use the model
+
+To generate text predictions, you need to download the model weights. **If you don't have them, check out our [guide](howto/download_weights.md).**
+
+Run inference:
+
+```bash
+python generate.py --prompt "Hello, my name is"
+```
+
+This will run the 7B model and require ~26 GB of GPU memory (A100 GPU).
+
+[Full guide for generating samples from the model](howto/inference.md).
+
+### Run Lit-StableLM on consumer devices
+
+On GPUs with `bfloat16` support, the `generate.py` script will automatically convert the weights and consume about ~14 GB.
+For GPUs with less memory, or ones that don't support `bfloat16`, enable quantization (`--quantize llm.int8`):
+
+```bash
+python generate.py --quantize llm.int8 --prompt "Hello, my name is"
+```
+
+See `python generate.py --help` for more options.
+
+[Full guide for generating samples from the model](howto/inference.md).
+
+## Finetune the model
+
+Porting from Lit-LLaMA in progress 👷
+
+## Pre-training
+
+Porting from Lit-LLaMA in progress 👷
+
+## Get involved!
+
+We are on a quest towards fully open source AI.
+
+<img align="right" src="https://pl-public-data.s3.amazonaws.com/assets_lightning/Lit_LLaMA_Illustration3x.png" alt="Lit-LLaMA" width="128"/>
+
+Join us and start contributing, especially on the following areas:
+
+- [ ] [Pre-training](https://github.com/Lightning-AI/lit-llama/labels/pre-training)
+- [ ] [Fine-tuning (full and LoRA)](https://github.com/Lightning-AI/lit-llama/labels/fine-tuning)
+- [ ] [Quantization](https://github.com/Lightning-AI/lit-llama/labels/quantization)
+- [ ] [Sparsification](https://github.com/Lightning-AI/lit-llama/labels/sparsification)
```

Comment:
Is this list still worth having? Most have been addressed already

Reply:
True but it's good call to action (there's always more to)